### PR TITLE
First batch of module tmpl codestyle

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

First batch of module tmpl codestyle

### Testing Instructions

Please test and review the following frontend modules using the testing sampledata

- mod_articles_archive
- mod_articles_categories
- mod_articles_category
- mod_articles_latest
- mod_articles_news
- mod_articles_popular
- mod_banners
- mod_breadcrumbs
- mod_custom

### Expected result

Still works

### Actual result

Works with codestyle problems

### Documentation Changes Required

none

### CODE REVIEW

File: modules/mod_banners/tmpl/default.php [hunk 1]
```diff
@@ -10,105 +10,104 @@
...
+								/>
+							</a>
+						<?php else : ?>
+							<?php // Open in parent window ?>
+							<a
+								href="<?php echo $link; ?>"
+								title="<?php echo htmlspecialchars($item->name, ENT_QUOTES, 'UTF-8'); ?>">
+								<img
+									src="<?php echo $baseurl . $imageurl; ?>"
+									alt="<?php echo $alt;?>"
```
Comment (Quy) [inline modules/mod_banners/tmpl/default.php]: Add space before `?>"`. Do the same for the following 2 lines.

File: modules/mod_banners/tmpl/default.php [hunk 2]
```diff
@@ -10,105 +10,104 @@
...
+						<?php elseif ($target == 2) : ?>
+							<?php // Open in a popup window ?>
+							<a
+								href="<?php echo $link; ?>" onclick="window.open(this.href, '',
+									'toolbar=no,location=no,status=no,menubar=no,scrollbars=yes,resizable=yes,width=780,height=550');
+									return false"
+								title="<?php echo htmlspecialchars($item->name, ENT_QUOTES, 'UTF-8'); ?>">
+								<img
+									src="<?php echo $baseurl . $imageurl; ?>"
+									alt="<?php echo $alt;?>"
```
Comment (Quy) [inline modules/mod_banners/tmpl/default.php]: Add space before `?>"`. Do the same for the following 2 lines.

File: modules/mod_banners/tmpl/default.php [hunk 3]
```diff
@@ -10,105 +10,104 @@
...
+						<?php $target = $params->get('target', 1); ?>
+						<?php if ($target == 1) : ?>
+							<?php // Open in a new window ?>
+							<a
+								href="<?php echo $link; ?>" target="_blank" rel="noopener noreferrer"
+								title="<?php echo htmlspecialchars($item->name, ENT_QUOTES, 'UTF-8'); ?>">
+								<img
+									src="<?php echo $baseurl . $imageurl; ?>"
+									alt="<?php echo $alt;?>"
+									<?php if (!empty($width)) echo ' width="' . $width . '"';?>
```
Comment (Quy) [inline modules/mod_banners/tmpl/default.php]: Add space before `?>"`. Do the same for the following 2 lines.

File: modules/mod_articles_news/tmpl/_item.php [hunk 4]
```diff
@@ -22,21 +22,17 @@
...
-
 <?php if ($params->get('show_introtext', '1')) : ?>
 	<?php echo $item->introtext; ?>
 <?php endif; ?>
-
 <?php echo $item->afterDisplayContent; ?>
-
 <?php if (isset($item->link) && $item->readmore != 0 && $params->get('readmore')) : ?>
-	<?php echo '<a class="readmore" href="' . $item->link . '">' . $item->linkText . '</a>'; ?>
+	<a class="readmore" href="'<?php echo $item->link; ?>'">
```
Comment (Quy) [inline modules/mod_articles_news/tmpl/_item.php]: Remove `'` before and after the php code

File: modules/mod_banners/tmpl/default.php [hunk 5]
```diff
@@ -10,105 +10,104 @@
...
 						<?php if (!empty($height)) echo ' height="' . $height . '"';?>
-					/>
+					>
+						<param name="movie" value="<?php echo $imageurl; ?>" />
+						<embed
+							src="<?php echo $imageurl; ?>"
+							loop="false"
+							pluginspage="http://www.macromedia.com/go/get/flashplayer"
+							type="application/x-shockwave-flash"
+							<?php if (!empty($width)) echo ' width="' . $width . '"';?>
```
Comment (Quy) [inline modules/mod_banners/tmpl/default.php]: Add space before `?>`. Do the same below.

File: modules/mod_banners/tmpl/default.php [hunk 6]
```diff
@@ -10,105 +10,104 @@
...
-								src="<?php echo $baseurl . $imageurl; ?>"
-								alt="<?php echo $alt;?>"
-								<?php if (!empty($width)) echo ' width="' . $width . '"';?>
-								<?php if (!empty($height)) echo ' height="' . $height . '"';?>
-							/>
-						</a>
+						<?php // Just display the image if no link specified ?>
+						<img
+							src="<?php echo $baseurl . $imageurl; ?>"
+							alt="<?php echo $alt;?>"
```
Comment (Quy) [inline modules/mod_banners/tmpl/default.php]: Add space before `?>"`. Do the same for the following 2 lines.

**Comment by dgrammatiko** [comment]
@zero-24 you're killing us, do it in J4...

File: modules/mod_breadcrumbs/tmpl/default.php [hunk 7]
```diff
@@ -22,54 +22,50 @@
...
 					</span>
 				<?php endif; ?>
-
 				<?php if (($key !== $penult_item_key) || $show_last) : ?>
 					<span class="divider">
 						<?php echo $separator; ?>
 					</span>
 				<?php endif; ?>
-				<meta itemprop="position" content="<?php echo $key + 1; ?>">
+				<meta itemprop="position" content="<?php echo $key + 1; ?>" />
```
Comment (brianteeman) [inline modules/mod_breadcrumbs/tmpl/default.php:59]: you have added a closing / here but this is not required for html5

**Comment by mbabker** [comment]
> you're killing us, do it in J4...

It'd actually be better if the code style were consistent across branches.  If you're making massive code style changes in only one branch you're making even more merge conflict issues as time goes on.

**Comment by zero-24** [comment]
> It'd actually be better if the code style were consistent across branches. If you're making massive code style changes in only one branch you're making even more merge conflict issues as time goes on.

So what is your suggestion? Doing it against staging or 4.0?

**Comment by mbabker** [comment]
Keep going against staging.  In theory there should only be one merge that has major conflicts, every merge after that should only deal with lines changed between the two branches.

File: modules/mod_articles_categories/tmpl/default_items.php [hunk 8]
```diff
@@ -9,29 +9,29 @@
...
-		<?php echo $item->title; ?>
-			<?php if ($params->get('numitems')) : ?>
-				(<?php echo $item->numitems; ?>)
-			<?php endif; ?>
-		</a>
-   		</h<?php echo $params->get('item_heading') + $levelup; ?>>
-
+			<a href="<?php echo JRoute::_(ContentHelperRoute::getCategoryRoute($item->id)); ?>">
+				<?php echo $item->title; ?>
+					<?php if ($params->get('numitems')) : ?>
```
Comment (Quy) [inline modules/mod_articles_categories/tmpl/default_items.php]: Lines 18-20: Remove tab

File: modules/mod_articles_category/tmpl/default.php [hunk 9]
```diff
@@ -13,71 +13,65 @@
...
-						<?php endif; ?>
-
-						<?php if ($item->displayCategoryTitle) : ?>
-							<span class="mod-articles-category-category">
-								(<?php echo $item->displayCategoryTitle; ?>)
-							</span>
-						<?php endif; ?>
-
-						<?php if ($item->displayDate) : ?>
-							<span class="mod-articles-category-date"><?php echo $item->displayDate; ?></span>
```
Comment (Quy) [inline modules/mod_articles_category/tmpl/default.php]: Multiple lines

File: modules/mod_articles_category/tmpl/default.php [hunk 10]
```diff
@@ -13,71 +13,65 @@
...
+						<li>
+							<?php if ($params->get('link_titles') == 1) : ?>
 								<a class="mod-articles-category-title <?php echo $item->active; ?>" href="<?php echo $item->link; ?>">
-									<?php if ($item->params->get('access-view') == false) : ?>
-										<?php echo JText::_('MOD_ARTICLES_CATEGORY_REGISTER_TO_READ_MORE'); ?>
-									<?php elseif ($readmore = $item->alternative_readmore) : ?>
-										<?php echo $readmore; ?>
-										<?php echo JHtml::_('string.truncate', $item->title, $params->get('readmore_limit')); ?>
-											<?php if ($params->get('show_readmore_title', 0) != 0) : ?>
-												<?php echo JHtml::_('string.truncate', $this->item->title, $params->get('readmore_limit')); ?>
```
Comment (Quy) [inline modules/mod_articles_category/tmpl/default.php]: Lines 66-68: Don't indent?

File: modules/mod_articles_category/tmpl/default.php [hunk 11]
```diff
@@ -13,71 +13,65 @@
...
+								<span class="mod-articles-category-date"><?php echo $item->displayDate; ?></span>
+							<?php endif; ?>
+							<?php if ($params->get('show_introtext')) : ?>
+								<p class="mod-articles-category-introtext">
+									<?php echo $item->displayIntrotext; ?>
+								</p>
+							<?php endif; ?>
+							<?php if ($params->get('show_readmore')) : ?>
+								<p class="mod-articles-category-readmore">
+									<a class="mod-articles-category-title <?php echo $item->active; ?>" href="<?php echo $item->link; ?>">
```
Comment (Quy) [inline modules/mod_articles_category/tmpl/default.php:57]: Multiple lines

Comment (zero-24) [reply modules/mod_articles_category/tmpl/default.php:57]: hmm can you please doube check that? Looks good to me?

File: modules/mod_articles_category/tmpl/default.php [hunk 12]
```diff
@@ -13,111 +13,101 @@
...
+									<?php echo $item->displayAuthorName; ?>
+								</span>
+							<?php endif; ?>
+							<?php if ($item->displayCategoryTitle) : ?>
+								<span class="mod-articles-category-category">
+									(<?php echo $item->displayCategoryTitle; ?>)
+								</span>
+							<?php endif; ?>
+							<?php if ($item->displayDate) : ?>
+								<span class="mod-articles-category-date"><?php echo $item->displayDate; ?></span>
```
Comment (Quy) [inline modules/mod_articles_category/tmpl/default.php]: Multiple lines.

File: modules/mod_custom/tmpl/default.php [hunk 13]
```diff
@@ -8,9 +8,8 @@
  */
 
 defined('_JEXEC') or die;
-?>
-
 
+?>
 <div class="custom<?php echo $moduleclass_sfx; ?>" <?php if ($params->get('backgroundimage')) : ?> style="background-image:url(<?php echo $params->get('backgroundimage'); ?>)"<?php endif; ?> >
```
Comment (Quy) [inline modules/mod_custom/tmpl/default.php]: Remove space `> >`

File: modules/mod_articles_categories/tmpl/default_items.php [hunk 14]
```diff
@@ -9,29 +9,29 @@
 
 defined('_JEXEC') or die;
 
-foreach ($list as $item) : ?>
+?>
+<?php foreach ($list as $item) : ?>
 	<li <?php if ($_SERVER['REQUEST_URI'] === JRoute::_(ContentHelperRoute::getCategoryRoute($item->id))) echo ' class="active"'; ?>> <?php $levelup = $item->level - $startLevel - 1; ?>
```
Comment (Quy) [inline modules/mod_articles_categories/tmpl/default_items.php]: Remove space after `<li `.
Move the second `<?php` next line.

File: modules/mod_articles_category/tmpl/default.php [hunk 15]
```diff
@@ -13,111 +13,103 @@
...
-						<?php if ($params->get('show_introtext')) : ?>
-							<p class="mod-articles-category-introtext">
-								<?php echo $item->displayIntrotext; ?>
-							</p>
-						<?php endif; ?>
-
-						<?php if ($params->get('show_readmore')) : ?>
-							<p class="mod-articles-category-readmore">
+			<li>
+				<div class="mod-articles-category-group"><?php echo $group_name; ?></div>
```
Comment (Quy) [inline modules/mod_articles_category/tmpl/default.php]: Multiple lines?

File: modules/mod_articles_latest/tmpl/default.php [hunk 16]
```diff
@@ -10,13 +10,13 @@
 defined('_JEXEC') or die;
 ?>
```
Comment (Quy) [inline modules/mod_articles_latest/tmpl/default.php]: Add blank line above. 

Is it possible to add lines when editing a file using Github web interface?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d46f2c12c736257215a009686a0daab307bb4a6b<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17851">issues.joomla.org/tracker/joomla-cms/17851</a>.</sub>

**Pull request closed without merging**